custom_components/vesync/humidifier.py

```diff
@@ -102,10 +101,11 @@ class VeSyncHumidifierHA(VeSyncDevice, HumidifierEntity):
     _attr_max_humidity = MAX_HUMIDITY
     _attr_min_humidity = MIN_HUMIDITY
 
-    def __init__(self, humidifier: VeSyncHumid200300S, coordinator) -> None:
+    def __init__(self, humidifier, coordinator) -> None:
         """Initialize the VeSync humidifier device."""
         super().__init__(humidifier, coordinator)
-        self.smarthumidifier = humidifier
+        if type(humidifier) in [VeSyncHumid200300S, VeSyncSuperior6000S]:
+            self.smarthumidifier = humidifier
```

Comment:
since the type hint was removed. There really should be an else after the type check to inform/raise if the type passed is not valid. Otherwise, you will have constructed the object without the property smarthumidifier

Reply:
I'll fix this too

Reply:
Latest commit has the fix for this issue as well
Would love your feedback on my error messages and handling (I'm raising an exception to break the init if an incompatible model is found so that it can't be instantiated)